matrix-sdk-android/src/main/java/org/matrix/android/sdk/internal/session/sync/job/SyncThread.kt

```diff
@@ -107,7 +107,11 @@ internal class SyncThread @Inject constructor(private val syncTask: SyncTask,
             Timber.tag(loggerTag.value).d("Pause sync...")
             isStarted = false
             retryNoNetworkTask?.cancel()
-            syncScope.coroutineContext.cancelChildren()
+            // Do not cancel the current incremental sync.
+            // Incremental sync can be long and it requires the user to wait for the treatment to end,
+            // else all is restarted from the beginning each time the user moves the app to foreground.
+            Timber.tag(loggerTag.value).d("Pause sync... Not cancelling incremental sync")
+            // syncScope.coroutineContext.cancelChildren()
```

Comment:
Fix looks good but I'd rather remove the offending line than simply comment it out. We always have git history if we need to bring it back, but commented code is a code smell

Reply:
I agree on the principle, also sonar for instance complain about commented code. But git history is less visible than a commented out line of code. Also sometimes the history can be lost during rework (if not done correctly).
So I prefer to comment the line.
Also in case of trouble it's easier to see the change and to revert.

Reply:
I would also be in the _removing the line_ camp, if we want to protect against regressions a unit test to replace the comment which asserts that pauses do not cancel inprogress syncs would be stronger 

although with the upcoming rust SDK changes I'm not sure it would be the best use of time to add tests here~

if context/history is important I would be for linking to the PR rather than commenting the line

Reply:
OK. I have force push the last commit, so that it reduces noise. Also note that the commit contains the changelog file, which links to the PR :).